mithril-client/src/client.rs

```diff
@@ -0,0 +1,149 @@
+//! Client module
+//!
+//! The client wraps the tedious initialization of the Mithril Client services;
+//! It offers a simple API to use the underlying mechanics of the library.
+//!
+//! ```
+//! use mithril-client::client::Client;
+//! use mithril-client::common::*:
+//!
+//! #[tokio::main]
+//! async fn main() -> StdResult<()> {
+//!     let client = Client::new("WRITE THE VKEY HERE", &http_server.url()).await?;
+//!     let snapshot = client.show_snapshot(&snapshot_message.digest).await?;
+//!
+//!     println!("Snapshot id={}, size={}", snapshot.digest, snapshot.size);
+//!
+//!     Ok(())
+//! }
+//! ```
+//!
+use std::sync::Arc;
+
+use crate::dependencies::DependenciesBuilder;
+use anyhow::Context;
+use config::builder::DefaultState;
+use config::ConfigBuilder;
+use mithril_common::messages::MithrilStakeDistributionListMessage;
+use mithril_common::StdResult;
+
+use crate::common::*;
+use crate::services::{MithrilStakeDistributionService, SnapshotService};
+/// Client structure that instanciates required dependencies
+pub struct Client {
+    snapshot_service: Arc<dyn SnapshotService>,
+    mithril_stake_distribution_service: Arc<dyn MithrilStakeDistributionService>,
+}
+
+impl Client {
```

Comment:
Snapshot download is missing, is this intentional ?

Reply:
Yes, it hasn't been implemented yet because it requires more in-depth modifications. In particular point 3 of 'Later' section of the issue #1141.